### PR TITLE
Add github actions workflow

### PR DESCRIPTION
**Pull request by dorileo**
Preparing to migrate from travis to github actions.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+      with:
+        submodules: true
+    - name: Prepare
+      run: |
+        sudo apt-get update
+        sudo apt-get install libefiboot-dev check grep ninja-build meson gcc-5 libblkid-dev lcov valgrind gnu-efi libefivar-dev python3-pip ruby ruby-dev rubygems python3-setuptools
+        sudo gem install coveralls-lcov
```
Comment (bryteise) [inline .github/workflows/main.yml]: Is this still usable? I'm not sure how coveralls is working in this new world order.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+      with:
+        submodules: true
+    - name: Prepare
+      run: |
+        sudo apt-get update
+        sudo apt-get install libefiboot-dev check grep ninja-build meson gcc-5 libblkid-dev lcov valgrind gnu-efi libefivar-dev python3-pip ruby ruby-dev rubygems python3-setuptools
```
Comment (bryteise) [inline .github/workflows/main.yml]: Doesn't look like we need the python bits these days perhaps?

**Comment by bryteise** [comment]
Looks reasonable to me and if it works then hey, +1

File: .github/workflows/main.yml (hunk 1 above)
Comment (dorileo) [reply .github/workflows/main.yml]: yes, still working and it's required by the test suite - I can check later if we can get rid of it or not.

File: .github/workflows/main.yml (hunk 2 above)
Comment (dorileo) [reply .github/workflows/main.yml]: Yep, makes sense - I'll push a new version without it. thanks.

File: .github/workflows/main.yml (hunk 1 above)
Comment (bryteise) [reply .github/workflows/main.yml]: Ah seems fine, looking at the output.

File: .github/workflows/main.yml (hunk 2 above)
Comment (bryteise) [reply .github/workflows/main.yml]: Well setuptools and pip at least.

**Pull request merged**